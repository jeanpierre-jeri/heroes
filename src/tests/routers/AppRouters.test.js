import { AuthContext } from "../../auth/authContext";
import {mount} from 'enzyme'
import AppRouters from "../../routers/AppRouters";


describe('Pruebas en <AppRouter />', () => {

  test('debe de mostrar el login si no está autenticado', () => {
    
    const contextValue = {
      user: {
        logged: false
      }
    }

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouters />
      </AuthContext.Provider>
    )
    
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('h1').text().trim()).toBe('Login')

  });

  test('debe de mostrar el componente de Marvel si está autenticado', () => {
    
    const contextValue = {
      user: {
        logged: true,
        name: 'Juancho'
      }
    }

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouters/>
      </AuthContext.Provider>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.navbar').exists()).toBeTruthy()

  });
  
  

})