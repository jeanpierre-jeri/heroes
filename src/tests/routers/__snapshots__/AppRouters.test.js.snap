// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el componente de Marvel si está autenticado 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="navbar-item nav-link text-info"
        >
          Juancho
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <hr />
      <div
        className="row row-cols-1 row-cols-md-3 row-cols-lg-4 g-4 animate__animated animate__fadeIn"
      >
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Spider Man"
              className="card-img-top"
              src="/assets/marvel-spider.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Spider Man
              </h5>
              <p
                className="card-text"
              >
                Peter Parker
              </p>
              <p
                className="card-text"
              >
                Amazing Fantasy #15
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-spider"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Captain America"
              className="card-img-top"
              src="/assets/marvel-captain.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Captain America
              </h5>
              <p
                className="card-text"
              >
                Steve Rogers
              </p>
              <p
                className="card-text"
              >
                Captain America Comics #1
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-captain"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Iron Man"
              className="card-img-top"
              src="/assets/marvel-iron.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Iron Man
              </h5>
              <p
                className="card-text"
              >
                Tony Stark
              </p>
              <p
                className="card-text"
              >
                Tales of Suspense #39
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-iron"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Thor"
              className="card-img-top"
              src="/assets/marvel-thor.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Thor
              </h5>
              <p
                className="card-text"
              >
                Thor Odinson
              </p>
              <p
                className="card-text"
              >
                Journey into Myster #83
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-thor"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Hulk"
              className="card-img-top"
              src="/assets/marvel-hulk.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Hulk
              </h5>
              <p
                className="card-text"
              >
                Bruce Banner
              </p>
              <p
                className="card-text"
              >
                The Incredible Hulk #1
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-hulk"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Wolverine"
              className="card-img-top"
              src="/assets/marvel-wolverine.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Wolverine
              </h5>
              <p
                className="card-text"
              >
                James Howlett
              </p>
              <p
                className="card-text"
              >
                The Incredible Hulk #180
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-wolverine"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Daredevil"
              className="card-img-top"
              src="/assets/marvel-daredevil.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Daredevil
              </h5>
              <p
                className="card-text"
              >
                Matthew Michael Murdock
              </p>
              <p
                className="card-text"
              >
                Daredevil #1
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-daredevil"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Hawkeye"
              className="card-img-top"
              src="/assets/marvel-hawkeye.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Hawkeye
              </h5>
              <p
                className="card-text"
              >
                Clinton Francis Barton
              </p>
              <p
                className="card-text"
              >
                Tales of Suspense #57
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-hawkeye"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Cyclops"
              className="card-img-top"
              src="/assets/marvel-cyclops.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Cyclops
              </h5>
              <p
                className="card-text"
              >
                Scott Summers
              </p>
              <p
                className="card-text"
              >
                X-Men #1
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-cyclops"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
        <div
          className="col animate__animated animate__fadeIn"
        >
          <div
            className="card h-100"
            style={
              Object {
                "width": "18rem",
              }
            }
          >
            <img
              alt="Silver Surfer"
              className="card-img-top"
              src="/assets/marvel-silver.jpg"
            />
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Silver Surfer
              </h5>
              <p
                className="card-text"
              >
                Norrin Radd
              </p>
              <p
                className="card-text"
              >
                The Fantastic Four #48
              </p>
              <a
                className="btn btn-primary"
                href="/hero/marvel-silver"
                onClick={[Function]}
              >
                Ver más...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en <AppRouter /> debe de mostrar el login si no está autenticado 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
